code/code/cmd/cmd_bash.cc

```diff
@@ -94,162 +93,216 @@ bool TBeing::canBash(TBeing *victim, silentTypeT silent)
       sendTo("You can't bash someone that is flying unless you are also.\n\r");
     return FALSE;
   }
+
   if (riding) {
     if (!silent)
       sendTo("You can't bash while mounted!\n\r");
     return FALSE;
-  }
-  if (getHeight() > 3*victim->getHeight()) {
-    if (!silent)
-      sendTo("That creature is too short to bash properly.\n\r");
-    return FALSE;
-  }
-  if (3*getHeight() < 2*victim->getHeight()) {
-    if (!silent)
-      sendTo("You don't stand a chance of bashing somebody that tall!\n\r");
-    return FALSE;
-  }
+  }   
+
   if (victim->isImmortal()) {
     if (!silent)
       sendTo("You slam into them but seeing how they are immortal it does no good.\n\r");
     return FALSE;
   }
+
   if (victim->getPosition() <= POSITION_SITTING) {
     if (!silent)
       sendTo(format("How can you bash someone already on the %s?!?\n\r") % roomp->describeGround());
     return FALSE;
   }    
+
   if (getMove() < 5) {
     if (!silent)
       sendTo("You don't have the vitality to bash anyone!\n\r");
     return FALSE;
   }
+
   if (!victim->hasLegs()) {
     if (!silent)
       sendTo("You can't knock them over, they have no legs.\n\r");
     return FALSE;
   }
 
+  if ((isSwimming() || victim->isSwimming())) {
```

Comment:
Moved this check into canBash() just to be consistent. I was considering making bashing while swimming for aquatic races (so really only Kalysians, as far as PC races go) but wasn't sure it was worth it, since it's such an edge case and would require updating the crashLanding method as well.